### ISSUE TITLE
Setting the host and port for workers.

### ISSUE THREAD

**Issue by jbotaldrich** [uber/cadence-java-client#239]
I've been trying to set the host and port for my workers to poll to a remote server.  I've tried creating my own WorkflowServiceTChannel with the appropriate host and port settings as well as setting these properties when doing "new Worker.Factory(host, port, domain)"

I noticed it kept defaulting to localhost as I would bring them up.  When I did a little debugging I noticed all the methods in WorkflowServiceTChannel use doRemoteCAll which uses the SubChannel.send(request).  This is an overloaded method for send(request, host, port) which passes null, and 0 for host and port respectively.  Is this the intention or is there some other way to set this that I've been missing?

**Comment by mfateev** [comment]
This sounds like a bug. Internally we use a different way to initialize the WorkflowService, so this code path might be untested.

**Comment by jbotaldrich** [comment]
Ignore this, I realized in starting up the cadence orchestrator on aws I hadn't registered the domain, (I started using the samples as a template and forgot about this step) .  The error I got though was that an Unknown Error had occurred. I went down the wrong path for a while until I realized the domain registration thing.  Might be useful to have something more descriptive for an error in that case.  Thanks anyways!
